From 87fdfbf943b7cde4462eac9083f46fc6c7192f05 Mon Sep 17 00:00:00 2001
From: Piotr Pawliczek <pawliczek@chromium.org>
Date: Fri, 14 Sep 2018 08:58:07 -0700
Subject: Patch ghostscript-9.20-runlibfileifexists.patch rebased
 to version 9.25. It includes two following fixes:

 1. Make sure customizations to /etc/ghostscript/Fontmap.local are used

  For more info, see: https://bugzilla.redhat.com/show_bug.cgi?id=610301

 2. Fix error when using ImageMagick's convert:

  Error: /undefined in .runlibfileexists
---
 Resource/Init/Fontmap     |  2 ++
 Resource/Init/gs_fonts.ps | 15 ++++++++++++---
 Resource/Init/gs_init.ps  |  9 +++++++++
 3 files changed, 23 insertions(+), 3 deletions(-)

diff --git a/Resource/Init/Fontmap b/Resource/Init/Fontmap
index ae5af96e0..72aeb1fa3 100644
--- a/Resource/Init/Fontmap
+++ b/Resource/Init/Fontmap
@@ -2,3 +2,5 @@
 % See Fontmap.GS for the syntax of real Fontmap files.
 %% Replace 1 (Fontmap.GS)
 (Fontmap.GS) .runlibfile
+% must be at the bottom of line to allow people overriding everything.
+(Fontmap.local) .runlibfileifexists
diff --git a/Resource/Init/gs_fonts.ps b/Resource/Init/gs_fonts.ps
index 38f0f6ca2..46c36e7d1 100644
--- a/Resource/Init/gs_fonts.ps
+++ b/Resource/Init/gs_fonts.ps
@@ -90,10 +90,19 @@ userdict /.nativeFontmap .FontDirectory maxlength dict put
                 % stack: dict file fontname filename|aliasname
       1 index type /stringtype eq
       1 index type /nametype eq and 1 index xcheck and
-      1 index /run eq 2 index /.runlibfile eq or and {
+      1 index /run eq 2 index /.runlibfile eq 3 index /.runlibfileifexists eq or or and {
                 % This is an inclusion entry.
-        pop findlibfile { exch pop } { file } ifelse
-        2 index exch .readFontmap pop
+        0 index /.runlibfileifexists eq {
+          pop findlibfile {
+            exch pop
+            2 index exch .readFontmap pop
+          } {
+            pop
+          } ifelse
+        } {
+          pop findlibfile { exch pop } { file } ifelse
+          2 index exch .readFontmap pop
+        } ifelse
       } {
                 % This is a real entry.
                 % Read and pop tokens until a semicolon.
diff --git a/Resource/Init/gs_init.ps b/Resource/Init/gs_init.ps
index bec307d1b..76544c01a 100644
--- a/Resource/Init/gs_init.ps
+++ b/Resource/Init/gs_init.ps
@@ -704,6 +704,14 @@ systemdict /internaldict dup .makeinternaldict .makeoperator
            { /undefinedfilename signalerror }
           ifelse
         } bind def
+/runlibfileifexists
+        {		% We don't want to bind 'run' into this procedure,
+                        % since run may get redefined.
+          findlibfile
+           { exch pop /run .systemvar exec }
+           { pop }
+          ifelse
+        } bind def
 /selectdevice
         { finddevice setdevice .setdefaultscreen } bind def
 /signalerror		% <object> <errorname> signalerror -
@@ -855,6 +863,7 @@ userdict /.currentresourcefile //null put
         } bind def
 % Temporarily substitute it for the real runlibfile.
 /.runlibfile /runlibfile load def
+/.runlibfileifexists /runlibfileifexists load def
 /runlibfile /runlibfile0 load def
 
 % Create the error handling machinery.
-- 
2.19.0.397.gdd90340f6a-goog

